MinGW Linux-to-Windows build broken by unnecessary capitalization

The MinGW-w64 Linux host to Windows 64bit binary build is broken because miniz added this include with a capital W:
```
/tmp/repository/vendor/miniz/include/miniz/miniz.c:3002:10: fatal error: Windows.h: No such file or directory
 3002 | #include <Windows.h>
      |          ^~~~~~~~~~~
compilation terminated.
```
While this doesn't matter when building on a MS Windows host with its case insensitive filenames, on a Linux host with MinGW it needs to be lowercase or it won't work. So it'd be cool if this could be fixed, because this is quite the showstopper sadly.

--- a/miniz_zip.c
+++ b/miniz_zip.c
@@ -42,7 +42,7 @@ extern "C" {
 #if defined(_MSC_VER) || defined(__MINGW64__)
 
 #define WIN32_LEAN_AND_MEAN
-#include <Windows.h>
+#include <windows.h>
 
 static WCHAR* mz_utf8z_to_widechar(const char* str)
 {

